getconf/base.py

```diff
@@ -258,7 +258,11 @@ def getint(self, key, default=0, doc=''):
         value = self._get(key, default=default, doc=doc, type_hint='int')
         if value is None:
             return None
-        return int(value)
+        try:
+            return int(value)
+        except ValueError:
+            raise ValueError(
+                "Unable to cast '%s' as integer for the key %s." % (value, key))
```

Comment:
We could use `repr` and remove the quotes around "%s". Numeric values would not be displayed the same as  strings, which would be a bit clearer:

    "Unable to cast %s as integer for the key %s." % (repr(value), key))